front/components/AssistantBuilderDataSourceModal.tsx

```diff
@@ -0,0 +1,263 @@
+import {
+  CloudArrowDownIcon,
+  IconButton,
+  Item,
+  Modal,
+  PageHeader,
+  TrashIcon,
+  XCircleIcon,
+} from "@dust-tt/sparkle";
+import { Transition } from "@headlessui/react";
+import type * as React from "react";
+import { useEffect, useState } from "react";
+
+import DataSourceResourceSelectorTree from "@app/components/DataSourceResourceSelectorTree";
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import { ConnectorProvider } from "@app/lib/connectors_api";
+import { DataSourceType } from "@app/types/data_source";
+import { WorkspaceType } from "@app/types/user";
+
+export default function AssistantBuilderDataSourceModal({
+  isOpen,
+  setOpen,
+  owner,
+  dataSources,
+  onSave,
+  dataSourceToManage,
+  onDelete,
+}: {
+  isOpen: boolean;
+  setOpen: (isOpen: boolean) => void;
+  owner: WorkspaceType;
+  dataSources: DataSourceType[];
+  onSave: (
+    dataSource: DataSourceType,
+    selectedResources: Record<string, string>
+  ) => void;
+  dataSourceToManage: {
+    dataSource: DataSourceType;
+    selectedResources: Record<string, string>;
+  } | null;
+  onDelete?: () => void;
+}) {
+  const [selectedDataSource, setSelectedDataSource] =
+    useState<DataSourceType | null>(null);
+  const [selectedResources, setSelectedResources] = useState<
+    Record<string, string>
+  >({});
+
+  useEffect(() => {
+    if (dataSourceToManage) {
+      setSelectedDataSource(dataSourceToManage.dataSource);
+      setSelectedResources(dataSourceToManage.selectedResources);
+    }
+  }, [dataSourceToManage]);
+
+  const onClose = () => {
+    setOpen(false);
+    setTimeout(() => {
```

Comment:
Timeout is useful so you don't see the modal's state reset during the modal close animation